From c970b4f314131c7d20379cf84abef09e5e7d1019 Mon Sep 17 00:00:00 2001
From: Dudemanguy <random342@airmail.cc>
Date: Thu, 17 Mar 2022 19:39:34 -0500
Subject: [PATCH] fix building without systemd

sd-escape.c and sd-escape.h were skipped when building without systemd,
but they contain functions needed for xdg-desktop-portal to build.
Fortunately, neither file actually requires anything specific to systemd
so they can just always be included as part of the sources.
---
 src/Makefile.am.inc | 5 -----
 src/xdp-utils.c     | 2 +-
 2 files changed, 1 insertion(+), 6 deletions(-)

diff --git a/src/Makefile.am.inc b/src/Makefile.am.inc
index 1b93d92..39dbaf1 100644
--- a/src/Makefile.am.inc
+++ b/src/Makefile.am.inc
@@ -125,14 +125,9 @@ xdg_desktop_portal_SOURCES = \
 	src/flatpak-instance.h          \
 	src/portal-impl.h		\
 	src/portal-impl.c		\
-	$(NULL)
-
-if HAVE_LIBSYSTEMD
-xdg_desktop_portal_SOURCES += \
 	src/sd-escape.c		\
 	src/sd-escape.h		\
 	$(NULL)
-endif
 
 if HAVE_PIPEWIRE
 xdg_desktop_portal_SOURCES += \
diff --git a/src/xdp-utils.c b/src/xdp-utils.c
index 5017af9..5d0f44d 100644
--- a/src/xdp-utils.c
+++ b/src/xdp-utils.c
@@ -33,13 +33,13 @@
 
 #ifdef HAVE_LIBSYSTEMD
 #include <systemd/sd-login.h>
-#include "sd-escape.h"
 #endif
 
 #include <gio/gio.h>
 #include <gio/gunixoutputstream.h>
 #include <gio/gdesktopappinfo.h>
 
+#include "sd-escape.h"
 #include "xdp-utils.h"
 
 #define DBUS_NAME_DBUS "org.freedesktop.DBus"
-- 
2.35.1

